mersenne-31/src/mersenne_31.rs

```diff
@@ -90,6 +91,25 @@ impl Distribution<Mersenne31> for Standard {
     }
 }
 
+impl Serialize for Mersenne31 {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        // No need to convert to canonical.
+        serializer.serialize_u32(self.value)
+    }
+}
+
+impl<'a> Deserialize<'a> for Mersenne31 {
+    fn deserialize<D: Deserializer<'a>>(d: D) -> Result<Self, D::Error> {
+        let val = u32::deserialize(d)?;
+        // Ensure that `val` came from a valid field element
```

Comment:
nit: I'd add more clarification like "Ensure that `val` satisfies our invariant, i.e. it need not be canonical but must fit in 31 bits."